BDArmory/Weapons/Missiles/MissileLauncher.cs

```diff
@@ -1748,6 +1788,11 @@ IEnumerator BoostRoutine()
                     }
 
                 //thrust
+                if (useFuel && burnRate > 0 && burnedMass < boosterMass)
+                {
+                    burnedMass = Mathf.Min(burnedMass + burnRate, burnedMass);
+                    ordinanceMass = burnedMass;
```

Comment:
burnRate is positive (boostermass / boostTime * 0.02), which means that burnedMass = mathf.Min is always going to return burnedMass, not burnedMass + burnrate; why is this Mathf.Min here?
burnedMass also hasn't been declared yet, so would be yielding either a null value or 0 at this point, so your ordinanceMass would be getting set to 0 and no fuelburn would occur.
Additionally, ordinanceMass, as an IPartmassModifier value, is adding its value to the base part.mass; burnedMass needs to be a negative number, or ordinanceMass = burnedMass would result in the missile getting heavier as fuel burns,
As per my previous comment, this should be ``ordinanceMass -= BurnRate;``, to have the fuel mass decremented by the burnRate every physics tick.
You could also get rid of burnedMass completely, and do ``if (useFuel && burnRate > 0 && -ordinanceMass < boosterMass), as oridianceMass up to this point should be a negative number between 0 and -boosterMass.

Reply:
The `burnedMass` on the right ought to be `boosterMass` from this comment above https://github.com/BrettRyland/BDArmory/pull/517#discussion_r1195661463, which prevents the `burnedMass` from exceeding the `boosterMass`. The purpose of calculating `burnedMass` and then setting `ordinanceMass` is to avoid exceeding `boosterMass`.

Reply:
ah, fair; yeah, having it be Mathf.Min(burnedmass + burnRate, *boosterMass*) makes much more sense to have there.

Reply:
this was a typo mistake gonna fix it now